SocketRocket/SRWebSocket.m

```diff
@@ -997,7 +1014,9 @@ - (void)_readFrameContinue;
 - (void)_readFrameNew;
 {
     dispatch_async(_workQueue, ^{
-        [_currentFrameData setLength:0];
+        // Don't reset the length, since Apple doesn't guarantee that this will free the memory (and in tests on
+        // some platforms, it doesn't seem to, effectively causing a leak the size of the biggest frame so far).
+        _currentFrameData = [[NSMutableData alloc] init];
```

Comment:
This by itself is worth a ton!
Indeed this is the only valid way to reset the data...